app/models/update_log.rb

```diff
@@ -1,17 +1,63 @@
 class UpdateLog
+  ####################
+  # CONSTANTS        #
+  ####################
+  MAX_UPDATES_TO_KEEP = 10
 
-  def self.setting_record
-    Setting.find_or_create_by(key: 'metrics_update')
+  def self.log_updates(starttime, endtime)
+    add_new_log(starttime, endtime)
+    delete_old_log
+    log_delay
+    save_setting
   end
 
-  def self.log_update(time)
-    setting = self.setting_record
-    setting.value["last_update"] = time
-    setting.save
+  def self.average_delay
+    setting_record
+    @setting.value['average_delay'] ? @setting.value['average_delay']: nil
   end
 
   def self.last_update
-    self.setting_record
-    Setting.find_by(key: 'metrics_update').value['last_update']
+    setting_record
+    @setting.value['constant_update'] ? @setting.value['constant_update'].values.last['end_time'] : nil
   end
+
+  class << self
+
+    def setting_record
+      @setting = Setting.find_or_create_by(key: 'metrics_update')
+    end
+    
+    def add_new_log(starttime, endtime)
+      setting_record
+      if @setting.value['constant_update'] === nil
+        @setting.value['constant_update'] = Hash.new
+      end
+      @last_update = @setting.value['constant_update'].keys.last
+      if @last_update === nil
+        index = 0 
+      else
+        index = @last_update + 1
+      end
+      @setting.value['constant_update'][index] = {'start_time' => starttime, 'end_time' => endtime}
+    end
+
+    def delete_old_log
+      if @last_update != nil && @last_update >= MAX_UPDATES_TO_KEEP
```

Comment:
The most idiomatic Ruby way to do this kind of thing is with a guard clause instead of an if statement:

```ruby
return if some_conditions
do_the_thing
```